pilot/pkg/xds/gen.go

```diff
@@ -79,8 +80,12 @@ func (s *DiscoveryServer) findGenerator(typeURL string, con *Connection) model.X
 	// some types to use custom generators - for example EDS.
 	g := con.proxy.XdsResourceGenerator
 	if g == nil {
-		// TODO move this to just directly using the resource TypeUrl
-		g = s.Generators["api"] // default to "MCP" generators - any type supported by store
+		if strings.HasPrefix(typeURL, "istio.io/debug/") {
+			g = s.Generators["event"]
```

Comment:
shouldn't this be `s.Generators[TypeDebug]`?

Reply:
We don't need `Generators[TypeDebug]` if the typeURL is `istio.io/debug` (no trailing slash).  If the type URL is `istio.io/debug/syncz`, we do need a generator, but it is the event generator, not the debug one.